src/plugins/dashboard/server/usage/dashboard_telemetry.ts

```diff
@@ -10,151 +10,70 @@ import { ISavedObjectsRepository, SavedObjectAttributes } from 'src/core/server'
 import { EmbeddablePersistableStateService } from 'src/plugins/embeddable/common';
 import { SavedDashboardPanel730ToLatest } from '../../common';
 import { injectReferences } from '../../common/saved_dashboard_references';
-
-interface VisualizationPanel extends SavedDashboardPanel730ToLatest {
-  embeddableConfig: {
-    savedVis?: {
-      type?: string;
-    };
-  };
-}
-
-interface LensPanel extends SavedDashboardPanel730ToLatest {
-  embeddableConfig: {
-    attributes?: {
-      visualizationType?: string;
-      state?: {
-        visualization?: {
-          preferredSeriesType?: string;
-        };
-        datasourceStates?: {
-          indexpattern?: {
-            layers: Record<
-              string,
-              {
-                columns: Record<string, { operationType: string }>;
-              }
-            >;
-          };
+export interface DashboardCollectorData {
+  panels: {
+    total: number;
+    by_reference: number;
+    by_value: number;
+    by_type: {
+      [key: string]: {
+        total: number;
+        by_reference: number;
+        by_value: number;
+        details: {
+          [key: string]: number;
```

Comment:
The one nit I have here is that the details mapping is maybe too specific. If Lens for example, wanted to track something else, we'd be narrowing the type down too far. The schema would also need to be updated.

Reply:
Looked in to that, and it doesn't seem like it's possible with the current limitations on the `schema` object. Basically, the only types that are supported are:

![image](https://user-images.githubusercontent.com/8698078/163463231-7aa9b1fb-40a3-4610-972c-c45cbbb84f17.png)

I thought about making it `details: { [key: string]: unknown;` and giving the schema `{ type: object }` but that doesn't seem to be supported. @elastic/kibana-telemetry might be able to help us out here, but for now I think we have to stick to a flat JSON object for `details` 🤷

Reply:
Thanks for looking into this! Strange that only a subset of types are supported, but I suppose it makes sense if things are expected to mostly be counters. Leaving it as is sounds good to me!